### PR TITLE
Remove old 3rdparty QtJson code in favor of Qt5's QJsonDocument

### PR DESCRIPTION
**Pull request by ckamm**
For https://github.com/owncloud/client/issues/5710

### CODE REVIEW

**Comment by guruz** [comment]
Looks ok to me 👍  I've tested, connecting, activity view 

Maybe @SamuAlfageme could do a smoke test if adding a shibboleth account in wizard still works and if the client will display the user name in the tool bar for shibboleth. And sharing :)

BTW, maybe you could have avoided to change the API by using ```toVariantMap```. But it's all good now if it works, code looks fine

**Comment by ckamm** [comment]
Yep, affected areas are status.php, notifications (incoming remote shares etc.), sharing, shibboleth, activity view. I've tested all of them, but having another person look is certainly appreciated.

@guruz I thought about just using `toVariantMap` (I did that for capabilities, because that way it's easier to mock if necessary), but using the actual json should be more correct (for instance I'm not sure whether we can distinguish null and undefined otherwise).

File: src/gui/activitylistmodel.cpp [hunk 1]
```diff
@@ -152,17 +154,19 @@ void ActivityListModel::slotActivitiesReceived(const QVariantMap& json, int stat
     _currentlyFetching.remove(ast);
 
     foreach( auto activ, activities ) {
-        auto json = activ.toMap();
+        auto json = activ.toObject();
+
+        qDebug() << "ZZZ" << json;
```
Comment (SamuAlfageme) [inline src/gui/activitylistmodel.cpp]: @ckamm 💤? 😄

Comment (ckamm) [inline src/gui/activitylistmodel.cpp]: Oops, will get rid of that.

**Pull request merged**